Formula/mold.rb

```diff
@@ -20,6 +20,7 @@ class Mold < Formula
 
   on_macos do
     depends_on "llvm" => [:build, :test] if DevelopmentTools.clang_build_version <= 1200
+    depends_on macos: :monterey
```

Comment:
Why are we restricting this to Monterey and above?

Reply:
Because it failed to build on Big Sur, or so I thought?  You can try and relax it if you like; maybe I misread the logs.

Reply:
It should've been fixed and not merged, then. CC @chenrui333

Reply:
Works fine in #99538.

In the future, let's be more careful about adding restrictions to the macOS version to existing formulae, particularly when this requirement is not documented upstream. Adding these restrictions leads to a bad user experience when doing `brew upgrade`.